src/encode/n_axioms.ml

```diff
@@ -0,0 +1,5029 @@
+  ; apps T.IntSet [] %% []
+  ] %% []
+
+let intlit_distinct m n =
+  appb ~tys:[ t_idv ] B.Neq
+  [ apps (T.IntLit m) [] %% []
+  ; apps (T.IntLit n) [] %% []
+  ] %% []
+
+let intlit_zerocmp n =
+  if n <= 0 then
+    apps T.IntLteq
+    [ apps (T.IntLit n) [] %% []
+    ; apps (T.IntLit 0) [] %% []
+    ] %% []
+  else
+    apps T.IntLteq
+    [ apps (T.IntLit 0) [] %% []
+    ; apps (T.IntLit n) [] %% []
+    ] %% []
+
+let intplus_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntPlus
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ appb B.Conj
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntPlus
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intuminus_typing () =
+  quant Forall
+  [ "x" ] [ t_idv ]
+  ~pats:[ [
+    apps T.IntUminus
+    [ Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ apps T.Mem
+      [ Ix 1 %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntUminus
+        [ Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intminus_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntMinus
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ appb B.Conj
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntMinus
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let inttimes_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntTimes
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ appb B.Conj
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; apps T.Mem
+        [ Ix 1 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ] %% []
+    ; apps T.Mem
+      [ apps T.IntTimes
+        [ Ix 2 %% []
+        ; Ix 1 %% []
+        ] %% []
+      ; apps T.IntSet [] %% []
+      ] %% []
+    ] %% []
+  ) %% []
+
+let intexp_typing () =
+  quant Forall
+  [ "x" ; "y" ] [ t_idv ; t_idv ]
+  ~pats:[ [
+    apps T.IntExp
+    [ Ix 2 %% []
+    ; Ix 1 %% []
+    ] %% []
+  ] ]
+  ( appb B.Implies
+    [ List (And,
+      [ apps T.Mem
+        [ Ix 2 %% []
+        ; apps T.IntSet [] %% []
+        ] %% []
+      ; appb ~tys:[ t_idv ] B.Neq
+        [ Ix 2 %% []
+        ; apps (T.IntLit 0) [] %% []
+        ] %% []
```

Comment:
The documentation says: "We define a^b (exponentiation) for a > 0, or b > 0, or a =/= 0 and b ∈ Int, by the four axioms (...)"

The definition is not important here. Assuming a and b are integers, it seems like (a =/= 0) \/ (b > 0) is a sufficient condition. But (b >= 0) is not sufficient, just because 0^0 is unspecified.

I think I should change the condition to (a =/= 0) \/ (b > 0), but it feels a bit wrong.